common/functions/src/scalars/maths/pow.rs

```diff
@@ -0,0 +1,116 @@
+use common_datavalues::DataType;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use num::traits::Pow;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct PowFunction {
+    display_name: String,
+}
+
+impl PowFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(PowFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for PowFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn num_arguments(&self) -> usize {
+        2
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if is_numeric(&args[0]) || args[0] == DataType::String || args[0] == DataType::Null {
+            Ok(DataType::Float64)
+        } else {
+            Err(ErrorCode::IllegalDataType(format!(
+                "Expected numeric, but got {}",
+                args[0]
+            )))
+        }
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let x_column: &DataColumn = &columns[0].column().cast_with_type(&DataType::Float64)?;
+        let y_column: &DataColumn = &columns[1].column().cast_with_type(&DataType::Float64)?;
+
+        let result = match (x_column, y_column) {
+            (DataColumn::Array(x_series), DataColumn::Constant(y, _)) => {
+                if y.is_null() {
+                    DFFloat64Array::full_null(input_rows)
+                } else {
+                    let y: f64 = DFTryFrom::try_from(y.clone())?;
+                    let opt_iter = x_series.f64()?.into_iter().map(|x| x.map(|x| x.pow(y)));
+                    DFFloat64Array::new_from_opt_iter(opt_iter)
+                }
+            }
+            (DataColumn::Constant(x, _), DataColumn::Array(y_series)) => {
+                if x.is_null() {
+                    DFFloat64Array::full_null(input_rows)
+                } else {
+                    let x: f64 = DFTryFrom::try_from(x.clone())?;
+                    let opt_iter = y_series.f64()?.into_iter().map(|y| y.map(|y| x.pow(y)));
+                    DFFloat64Array::new_from_opt_iter(opt_iter)
+                }
+            }
+            _ => {
+                let x_series = x_column.to_minimal_array()?;
+                let y_series = y_column.to_minimal_array()?;
+                let opt_iter = x_series
+                    .f64()?
+                    .into_iter()
+                    .zip(y_series.f64()?.into_iter())
+                    .map(|(x, y)| match (x, y) {
```

Comment:
Try to use `binary` function in `ops/arith.rs`, this can reduce the match inside the loop.